src/components/AllMonsters.js

```diff
@@ -1,16 +1,37 @@
-import React from 'react'
+import React, { useState } from 'react'
 import styled from 'styled-components/macro'
 import LoadingAnimation from './LoadingAnimation'
 import OneMonsterItem from './OneMonsterItem'
+import Fuse from 'fuse.js'
 
 export default function AllMonsters({ loading, monsterDetails }) {
+  const [searchResults, setSearchResults] = useState('')
+
+  const fuse = new Fuse(monsterDetails, {
+    keys: ['name', 'type', 'challenge_rating'],
+  })
+  const results = searchResults ? fuse.search(searchResults) : monsterDetails
+
+  function handleInput({ currentTarget = {} }) {
+    const { value } = currentTarget
+    setSearchResults(value)
+  }
```

Comment:
I'd consider moving functions below the return of your components just for readability reasons.